contracts/helpers/TransferHelper.sol

```diff
@@ -70,97 +88,242 @@ contract TransferHelper is TransferHelperInterface, TokenTransferrer {
+
+                // Create a ConduitTransfer corresponding to each
+                // TransferHelperItem.
+                conduitTransfers[i] = ConduitTransfer(
+                    item.itemType,
+                    item.token,
+                    msg.sender,
+                    recipient,
+                    item.identifier,
+                    item.amount
+                );
+            }
         }
 
-        // Return a magic value indicating that the transfers were performed.
-        magicValue = this.bulkTransfer.selector;
+        // Check if the derived conduit is a contract.
+        if (!_isContract(conduit)) {
+            // If conduit is not a contract, revert with the passed in
+            // conduit key and derived conduit address.
+            revert InvalidConduit(conduitKey, conduit);
+        }
+
+        // Attempt the external call to transfer tokens via the derived conduit.
+        try ConduitInterface(conduit).execute(conduitTransfers) returns (
+            bytes4 conduitMagicValue
+        ) {
+            // Check if the value returned from the external call matches
+            // the conduit `execute` selector.
+            if (
+                conduitMagicValue != ConduitInterface(conduit).execute.selector
+            ) {
+                // If the external call fails, revert with the conduit key
+                // and conduit address.
+                revert InvalidMagicValue(conduitKey, conduit);
+            }
+        } catch (bytes memory data) {
+            // Catch reverts from the external call to the conduit and
+            // "bubble up" the conduit's revert reason if present.
+            if (data.length != 0) {
+                assembly {
+                    returndatacopy(0, 0, returndatasize())
+                    revert(0, returndatasize())
+                }
+            } else {
+                // If no revert reason is present, revert with a generic error
+                // with the conduit key and conduit address.
+                revert ConduitErrorGenericRevert(conduitKey, conduit);
+            }
+        } catch Error(string memory reason) {
+            // Catch reverts with a provided reason string and
+            // revert with the reason, conduit key and conduit address.
+            revert ConduitErrorString(reason, conduitKey, conduit);
+        } catch Panic(uint256 errorCode) {
+            // Catch reverts caused by a panic and revert with the
+            // panic error code, conduit key and conduit address.
+            revert ConduitErrorPanic(errorCode, conduitKey, conduit);
+        }
+    }
+
+    /**
+     * @notice Check if the passed-in address refers to a contract by comparing
+     *         its codeHash to the default codeHash for non-contract addresses.
+     *
+     * @param account The account to be checked.
+     */
+    function _isContract(address account) internal view returns (bool) {
```

Comment:
Can lose this if modifying as suggested above

Reply:
We’re also using `_isContract` in `_performTransfersWithoutConduit` to check if the recipient is a contract before calling `onERC721Received`. Should we replace with an alternate check or just call `onERC721Received` on all recipients?

Reply:
it looks like `_performERC721Transfer` and `_performERC1155Transfer` already do an extcodesize check:

https://github.com/ProjectOpenSea/seaport/blob/f62c14fe0a94fcb0d26ffdcee48238be63a85b4c/contracts/lib/TokenTransferrer.sol#L248-L253
if these contracts are not already loaded in memory, it would be cheaper to check with extcodehash rather than extcodesize (700 vs 300 gas)

<img width="285" alt="Screen Shot 2022-07-07 at 10 09 19 AM" src="https://user-images.githubusercontent.com/22116/177831399-c606ff3a-e9ff-4940-89a1-f630bf412315.png">

thus the instances above could also be optimized